.ci/magic-modules/generate-terraform.sh

```diff
@@ -19,19 +19,30 @@ ln -s "${PWD}/build/$SHORT_NAME/" "${GOPATH}/src/github.com/terraform-providers/
 popd
 
 pushd "${GOPATH}/src/github.com/terraform-providers/$PROVIDER_NAME"
+
 go get
+# This line removes every file which is not specified here.
+# If you add files to Terraform which are not generated, you have to add them here.
+# It uses the somewhat obtuse 'find' command.  To explain:
+# "find .": all files and directories recursively under the current directory, subject to matchers.
+# "-type f": all regular real files, i.e. not directories.
+# "-not": do the opposite of the next thing, always used with another matcher.
+# "-wholename": entire relative path - including directory names - matches following wildcard.
+# "-name": filename alone matches following string.  e.g. -name README.md matches ./README.md *and* ./foo/bar/README.md
+# "-exec": for each file found, execute the command following until the literal ';'
+find . -type f -not -wholename "./.git*" -not -wholename "./vendor*" -not -name ".travis.yml" -not -name "CHANGELOG.md" -not -name GNUmakefile -not -name LICENSE -not -name README.md -not -wholename "./examples*" -not -name "main.go" -not -wholename "./scripts*" -exec git rm {} \;
```

Comment:
It would be nice if you could split this list of files into a proper list. I'm not sure whether we'll add/remove things from this a lot but it would definitely make the list of ignored files a bit more readable.

Reply:
I took a quick stab and this is about the best I came up with:
```bash
ARRAY=(
  '-not -wholename "./.git*"'
  '-not -wholename "./vendor*"'
  '-not -name ".travis.yml"'
  '-not -name "CHANGELOG.md"'
  '-not -name GNUmakefile')

CMD="find . -type f ${ARRAY[*]}"
echo $CMD

eval $CMD
```
either way, not critical

Reply:
I think that programmatically constructing the list will be less readable, unfortunately.  I'll take it as a for-later task if this starts to be annoying to anyone.